R/get_source_expressions.R

```diff
@@ -152,6 +152,19 @@ lint_parse_error <- function(e, source_expression) {
   lint_parse_error_nonstandard(e, source_expression)
 }
 
+fixup_line <- function(line) {
+  # helper to ensure a valid string is provided as line where necessary handles tow cases:
+  #  * NA lines, as generated by running outside of code blocks in Rmd documents
+  #  * invalid string lines, as generated by invalid encoding settings
+  if (is.na(line)) {
+    return("")
+  }
+  tryCatch({
```

Comment:
Nit: for readability I might prefer s.t. like

```r
nchar_fails <- inherits(tryCatch(nchar(line, type = "chars"), error = identity), "error")
if (nchar_fails) {
  ""
} else {
  line
}
```

Or something along those lines. Now it might be tricky/require some mental cache to figure out what's actually being returned, and the construct of having two statements in the `tryCatch()` when only the first is expected to cause issues might be confusing (though I often use it myself and I'm sure it's used elsewhere in our sources...)

WDYT?

Reply:
Rewrote to 
```r
nchars <- tryCatch(nchar(line, type = "chars"), error = function(e) NA_integer_)
if (is.na(nchars)) {
  ""
} else {
  line
}
```